### PR TITLE
Adding decimal constraint

### PR DESCRIPTION
**Pull request by Aspoz**
This adresses the issue i filed #30, this will verify that an incoming parameters is checked for valid decimal.

Usage: Parameters.decimal(precision, scale)

### CODE REVIEW

File: lib/stronger_parameters/constraints/decimal_constraint.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    def value(v)
+      if scale > 0
+        if v.is_a?(Float) && (v.abs.to_s.size - 1) <= precision
+          return value(v.to_s)
+        elsif v.is_a?(String) && v =~ /\A-?\d{1,#{precision - scale}}\.?\d{#{scale}}\Z/
+          return v.to_d
+        end
+      elsif (v.is_a?(Fixnum) || v.is_a?(Bignum)) && (v.abs.to_s.size <= precision)
+        return v.to_d
+      elsif v.is_a?(String) && v =~ /\A-?\d{1,#{precision - scale}}\Z/
```
Comment (grosser) [inline lib/stronger_parameters/constraints/decimal_constraint.rb]: could .to_s and using this regex work for all cases ?

Comment (Aspoz) [inline lib/stronger_parameters/constraints/decimal_constraint.rb]: I tried to include a broad spectrum in my tests, I'm not sure if there could be other input possibilities. What I'm currently thinking though is that when you for instance have a 8/2 precision/scale, it isn't possible to accept integers. As it always is a valid decimal. What do you think?

Comment (grosser) [inline lib/stronger_parameters/constraints/decimal_constraint.rb]: yeah idk ... some might expect 1 to get converted to 1.00 ...

**Comment by Aspoz** [comment]
I delved a little deeper in what is accepted in a decimal.

And now when you have a 4/2 pre/sca, it will allow `1`/`1.2`/`1.23`/`12.3`/`12.34`. which according to my test with a psql db should be correct.

**Comment by grosser** [comment]
looks like rails 3.2 does not have `.to_d` :(

`NoMethodError: undefined method `to_d' for -12345:Fixnum`

**Comment by grosser** [comment]
this seems to work and pass travis:

```Ruby
require 'stronger_parameters/constraints'

module StrongerParameters
  class DecimalConstraint < Constraint
    def initialize(precision, scale)
      @precision = precision
      @scale = scale
      @regex = /\A-?\d{1,#{precision - scale}}#{"(\\.\\d{1,#{scale}})?" if scale > 0}\Z/
    end

    def value(v)
      match = v.to_s
      if match =~ @regex
        BigDecimal(match)
      else
        StrongerParameters::InvalidValue.new(v, "must be a decimal with precision #{@precision} and scale #{@scale}")
      end
    end
  end
end
```

**Comment by Aspoz** [comment]
Nice! I really like the this, going to change it.

**Comment by grosser** [comment]
thx, 2.3.0 is out!

**Pull request merged**